### ISSUE TITLE
pytorch lightning module log function raises error

### ISSUE THREAD

**Issue by gabriele-marino** [microsoft/pyright#10650]
**Describe the bug**
Pytorch Lightning log method raises a "partially unknown type" error, even though the method looks typed.

**Code or Screenshots**

```python
import pytorch_lightning as pl
from torch import Tensor


class MyModel(pl.LightningModule):

    def training_step(self, batch: int, batch_idx: int) -> Tensor:
        self.log("train_loss", 0.123)
        return Tensor([0.123])
```

  ... - error: Type of "log" is partially unknown
    Type of "log" is "(name: str, value: Metric | Tensor | int | float, prog_bar: bool = False, logger: bool | None = None, on_step: bool | None = None, on_epoch: bool | None = None, reduce_fx: str | ((...) -> Unknown) = "mean", enable_graph: bool = False, sync_dist: bool = False, sync_dist_group: Any | None = None, add_dataloader_idx: bool = True, batch_size: int | None = None, metric_attribute: str | None = None, rank_zero_only: bool = False) -> None" (reportUnknownMemberType)

**VS Code extension or command-line**
I am using command line pyright = "^1.1.402"

**Comment by erictraut** [comment]
By default, pyright doesn't generate any error for this code. I'm guessing that you've enabled strict-mode type checking or have manually enabled the `reportUnknownMemberType` diagnostic check. When this check is enabled, pyright will tell you when you access an attribute or method whose bound type is `Unknown` or partially `Unknown`. This is of interest to some users because `Unknown` can indicate that the type information is incomplete, which can lead to "holes" in type analysis.

If you look at the type of `log`, you'll see that its signature contains a type that is partially unknown. This comes from an incomplete type annotation on the `reduce_fx` parameter, which uses a bare `Callable` without supplying any type arguments.

So, pyright is working as intended here.

You may want to report the issue to the maintainers of this library.
